### PR TITLE
Publish scheduling reminder

### PR DESCRIPTION
**Pull request by planarvoid**
This PR connects the FluxC - https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/1309 and handles the notification shown when the user schedules a post. 

I'll add some more tests in another PR. I wanted to keep this one small.

Expected notification wording is: 
Title: `Scheduled post`
Body: `"[Post name]" has been published`

Title: `Scheduled post: 1 hour reminder`
Body: `"[Post name]" will be published in 1 hour`

Title: `Scheduled post: 10 minute reminder`
Body: `"[Post name]" will be published in 10 minutes` (edited)

To test:
* Go to a Post/Post settings/Publish settings
* Set publish time to the future (current time + 11 minutes)
* Click on "Notification" 
* Select "10 minutes before"
* Wait 2 minutes
* A notification is shown
* Repeat for the other 2 options

To test 2:
* Go to a Post/Post settings/Publish settings
* Set publish time to the future (current time + 11 minutes)
* Click on "Notification"
* Select "1 hour before"
* Click on "Notification"
* Select "10 minutes before"
* Wait 2 minutes
* A notification is shown (the previous notification was cancelled)
* Try with other options as well as setting Notification to "OFF"

Update release notes:

- [x] If there are user facing changes, I have added an item to `RELEASE-NOTES.txt`.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt [hunk 1]
```diff
@@ -161,6 +171,41 @@ class EditPostPublishSettingsViewModel
...
+    }
+
+    fun scheduleNotification(post: PostModel, notificationTime: SchedulingReminderModel.Period) {
+        updateNotifications(post, notificationTime)
+        updateUiModel(post)
+    }
+
+    private fun updateNotifications(
+        post: PostModel,
+        domainModel: SchedulingReminderModel.Period = OFF
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostPublishSettingsViewModel.kt]: Wonder, was this one supposed to be called something different than `domainModel` (eg. `notificationTime`?)

File: WordPress/src/main/java/org/wordpress/android/ui/posts/PublishNotificationReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                        ONE_HOUR -> string.notification_scheduled_post_one_hour_reminder to string.notification_post_will_be_published_in_one_hour
+                        TEN_MINUTES -> string.notification_scheduled_post_ten_minute_reminder to string.notification_post_will_be_published_in_ten_minutes
+                        WHEN_PUBLISHED -> string.notification_scheduled_post to string.notification_post_has_been_published
+                        OFF -> return
+                    }
+                    val title = resourceProvider.getString(titleRes)
+                    val message = resourceProvider.getString(messageRes, post.title)
+                    val notificationCompat = NotificationCompat.Builder(
+                            context,
+                            context.getString(R.string.notification_channel_normal_id)
```
Comment (khaykov) [inline WordPress/src/main/java/org/wordpress/android/ui/posts/PublishNotificationReceiver.kt]: I wonder, should we use `notification_channel_reminder_id` channel for notification like this?

Comment (planarvoid) [reply WordPress/src/main/java/org/wordpress/android/ui/posts/PublishNotificationReceiver.kt]: good point, I haven't noticed it's there

**Comment by planarvoid** [comment]
@khaykov thanks for checking. I've actually missed a TODO 😢 , it should all be fixed now

**Pull request merged**